tasks.c

```diff
@@ -3734,6 +3734,15 @@ void vTaskStartScheduler( void )
      * from getting optimized out as it is no longer used by the kernel. */
     ( void ) uxTopUsedPriority;
 
+    /* clean up so vTaskStartScheduler can be called again */
+    #if ( configNUMBER_OF_CORES == 1 )
+        pxCurrentTCB = NULL;
+    #else
+        memset(pxCurrentTCBs, 0, sizeof(pxCurrentTCBs));
```

Comment:
pxCurrentTCBs is a volatile variable. memset discards volatile qualifier from pointer target type [-Werror=discarded-qualifiers], resulting in error.
This would require writing out a loop, for setting the volatile array to 0

Reply:
fixed